pyccel/parser/semantic.py

```diff
@@ -1932,6 +1931,8 @@ def _get_indexed_type(self, base, args, expr):
                     class_type = HomogeneousTupleType
                 elif dtype_cls is PythonList:
                     class_type = HomogeneousListType
+                elif dtype_cls is PythonSet:
+                    class_type = HomogeneousSetType
```

Comment:
This is becoming quite a long list of if/elif blocks. Maybe a dict could be used here (this has already been mentioned in #1901 so it should probably be fixed there as that PR must be merged first anyway)

Reply:
this is now deprecated after I merged the `devel` branch into my branch